docs/rules/arrow-linebreak.md

```diff
@@ -0,0 +1,84 @@
+# Enforce the location of arrow function bodies with implicit returns (arrow-linebreak)
+
+An arrow function body can contain an implicit return as a single statement instead of a block. It can be useful to enforce a consistent location for the arrow function body.
+
+## Rule Details
+
+This rule aims to enforce a consistent location for single statement arrow function bodies.
+
+### Options
+
+This rule accepts a string option:
+
+* `"beside"` (default) disallows a newline before an arrow function body.
+* `"below"` requires a newline before an arrow function body.
+
+Examples of **incorrect** code for this rule with the default `"beside"` option:
+
+```js
+/* eslint arrow-linebreak: ["error", "beside"] */
+
+(foo) =>
+  bar;
+
+(foo) =>
+  (bar);
+
+(foo) =>
+  bar =>
+    baz;
+
+```
+
+Examples of **correct** code for this rule with the default `"beside"` option:
+
+```js
+/* eslint arrow-linebreak: ["error", "beside"] */
+
+(foo) => bar;
+
+(foo) => (bar);
+
+(foo) => bar => baz;
+
+(foo) => { // block statement function bodies are always allowed with this rule
+  return bar();
+}
+
+(foo) => ( // a function body with a leading parentheses is also always allowed with this rule
```

Comment:
What's the reasoning for always allowing this? I would have expected that an opening parenthesis on the first line would be treated the same way as anything else on the first line (i.e. allowed with `beside`, disallowed with `below`).

Reply:
Thanks for the review! I originally made this rule to be able to enforce that our code doesn't have lines that end in a `=>`, but I wanted to make an exception for cases like: 
```js
(foo) => (
  bar
);

// &

(foo) => 
  (
    bar
  );
```

However, I agree with you, treating them the same way will make this rule much clearer. The first case above should be allowed with `beside` and the second with `below`. I'll update the logic for this.